library/src/utils/text.ts

```diff
@@ -27,10 +21,19 @@ export const jsStrToObject = (raw: string) => {
   }
 }
 
-const caseFns: Record<string, (s: string) => string> = { kebab, snake, pascal }
+// The case mods expect the input to be raw attribute names (kebab-case)
+export const modCamel = (str: string) =>
+    str.replace(/-[a-z]/g, (x) => x[1].toUpperCase())
+
+export const modSnake = (str: string) => str.replace(/-/g, '_')
+
+export const modPascal = (str: string) =>
+    str[0].toUpperCase() + modCamel(str.slice(1))
+
+const caseFns: Record<string, (s: string) => string> = { camel: modCamel, snake: modSnake, pascal: modPascal }
 
-export function modifyCasing(str: string, mods: Modifiers) {
-  for (const c of mods.get('case') || []) {
+export function modifyCasing(str: string, mods: Modifiers, defaultCase: string = 'camel') {
+  for (const c of mods.get('case') || [defaultCase]) {
```

Comment:
Makes the default case a param so that only one case modifier is applied. Before the default case would be applied no matter what, then the user case applied on top of that.